core/validation/failures.py

```diff
@@ -147,10 +148,38 @@ def __str__(self):
             f'type "{self.expected_type}", got type "{self.actual_type}"'
         )
 
-    def to_user_centered_components(self) -> tuple[str, str, str, str]:
-        # Numbers - 1. Outcomes/Outputs is just numerical (e.g not m^2), 2. Funding/PSI is Monetary
-        # Dates - Programme Progress sheet: Start, Completion and Date of Most Important...
-        return "Unimplemented", "Unimplemented", "Unimplemented", "Unimplemented"
+    def to_user_centered_components(self) -> tuple[str, str, str]:
+        sheet = INTERNAL_TABLE_TO_FORM_TAB[self.sheet]
+        column, section = INTERNAL_COLUMN_TO_FORM_COLUMN_AND_SECTION[self.column]
+        expected_type = INTERNAL_TYPE_TO_USER_CENTERED_TYPE[self.expected_type]
+        actual_type = INTERNAL_TYPE_TO_USER_CENTERED_TYPE[self.actual_type]
+        if sheet == "Outcomes":
+            column, section = "F-O", "Project Outputs"
+
+        if self.expected_type == "datetime64[ns]":
+            message = (
+                f"For column {column} you entered {actual_type} when we expected {expected_type}. "
+                f"You must enter the start date in the correct format, for example, Dec-22, Jun-23"
+            )
+        else:
+            if sheet == "PSI":
+                message = (
+                    f"For column {column} you entered {actual_type} when we expected {expected_type} "
+                    f"You must enter the required data in the correct format, for example, £5,588.13 or £238,"
+                    f"062.50"
+                )
+            elif sheet in ["Funding Profiles", "Project Outputs", "Outcomes"]:
+                message = (
+                    f"Between columns {column} you entered {actual_type} when we expected {expected_type} "
+                    f"enter data using the correct format, for example, 9 rather than 9m2. Only use numbers"
+                )
+            else:
+                message = (
+                    f"For column {column} you entered {actual_type} when we expected {expected_type} "
+                    f"enter data using the correct format, for example, 9 rather than 9m2. Only use numbers"
+                )
+
```

Comment:
I'm adding a section name: "Outcome Indicators (excluding footfall) and Footfall Indicator"
so the returnwill read

```
"Outcome Indicators (excluding footfall) and Footfall Indicator)": [
          "Between columns E-W you entered text when we expected a number enter data using the correct format, for example, 9 rather than 9m2. Only use numbers"
        ]
```

Reply:
```
column, section = "F-O", "Outcome Indicators (excluding footfall) and Footfall Indicator"
```
Columns F-O doesn't align with the spreadsheet for football.. could be missleading

Reply:
I've expanded it to include the entire range of cells an error could occur, D-O. Not very satisfying I know but maybe the best I can do without making changes to the failure logic itself